app/controllers/gems_adoptions_controller.rb

```diff
@@ -20,6 +20,24 @@ def create
     end
   end
 
+  def update
+    adoption_request = AdoptionRequest.find(params[:id])
+    gems_adoption = GemsAdoption.find(adoption_request.gems_adoption_id)
+    gems_adoption.status = 'close'
+    if gems_adoption.save
+      gem_ownership_transfer = GemOwnershipTransfer.new(old_user_id: current_user.id, new_user_id: adoption_request.user_id, ruby_gem_id: gems_adoption.ruby_gem_id)
+      if gem_ownership_transfer.save
+        redirect_to gem_path(gems_adoption.ruby_gem)
+      else
+        flash[:success] = 'Gem couldnt be trespassed'
+        redirect_to adoption_requests_path
+      end
+    else
+      flash[:success] = 'Adoption request cant be accepted'
+      redirect_to adoption_requests_path
+    end
+  end
+
```

Comment:
I fact I think something like this would be a bit more readable:
```ruby
  ...

  def update
    adoption_request = AdoptionRequest.find(params[:id])
    gems_adoption = GemsAdoption.find(adoption_request.gems_adoption_id)
    gems_adoption.status = 'close'
    if gems_adoption.update_attributes(status: 'closed') && GemOwnershipTransfer.create(old_user_id: current_user.id, new_user_id: adoption_request.user_id, ruby_gem_id: gems_adoption.ruby_gem_id)
      redirect_to gem_path(gems_adoption.ruby_gem)
    else
      flash[:success] = "Gem couldn't be transferred"
       redirect_to adoption_requests_path
    end
  end

  ...
```

What do you think?